config/routes.rb

```diff
@@ -19,6 +19,7 @@
   resources :nodes, only: [:index, :show, :destroy] do
     collection do
       post :bootstrap
+      post :assign_roles
```

Comment:
On the route it's strange to see underscores, if I recall correctly this can be done using `post :assign_roles, path: 'assign-roles'`.